syntax = "proto3";

package fixbug;

//下面的选项，表示生成service服务类和rpc方法描述， 默认是不生成的
option cc_generic_services = true;

message ResultCode{  //定义返回的错误码
    int32 errcode = 1;
    bytes errmsg = 2;
}


//常用的数据类型： 数据、列表、映射表

//定义登录请求消息类型   name   pwd
message LoginRequest{
    bytes name = 1;   //等于1表示这是第一个参数，一般string的存储定义为bytes
    bytes pwd = 2;
   // map<int32,string> test = 3; //映射表类型
}

//定义登录响应消息类型
message LoginResponse{
    ResultCode result = 1;
    bool success = 2;
}

message RegisterRequest{
    bytes id = 1;
    bytes name = 2;   
    bytes pwd = 3;
}

message RegisterResponse{
    ResultCode result = 1;
    bool success = 2;
}


//定义远程rpc服务
service UserServiceRpc{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}